exporters/memory/test/in_memory_span_exporter_test.cc

```diff
@@ -0,0 +1,26 @@
+#include "opentelemetry/exporters/memory/in_memory_span_exporter.h"
+#include "opentelemetry/nostd/span.h"
+#include "opentelemetry/sdk/trace/span_data.h"
+
+#include <gtest/gtest.h>
+
+using opentelemetry::exporter::memory::InMemorySpanExporter;
+using opentelemetry::sdk::trace::Recordable;
+using opentelemetry::sdk::trace::SpanData;
+
+TEST(InMemorySpanExporter, ExportBatch)
+{
+  InMemorySpanExporter exporter;
+
+  ASSERT_EQ(0, exporter.GetData().get()->GetSpans().size());
+
+  opentelemetry::nostd::span<std::unique_ptr<Recordable>> batch(
+      new std::unique_ptr<Recordable>(new SpanData()), 1);
```

Comment:
This should rather be something along this lines:
```c++
std::unique_ptr<Recordable> spandata(new SpanData());
opentelemetry::nostd::span<std::unique_ptr<Recordable>> batch(&spandata, 1);
```
Otherwise we'll leak the memory of the dynamically allocated `unique_ptr` (`new std::unique_ptr<Recordable>`).

Reply:
Alright, updated